Makefile

```diff
@@ -468,9 +468,26 @@ test-packages:
 	cargo test -p wasmer-derive --release
 
 
-# The test-capi rules depend on the build-capi rules to build the .a files to
-# link the tests against. cargo test doesn't know that the tests will be running
-test-capi: $(foreach compiler_engine,$(compilers_engines),test-capi-$(compiler_engine))
+# We want to run all the tests for all available compilers. The C API
+# and the tests rely on the fact that one and only one default
+# compiler will be selected at compile-time. Therefore, if we want to
+# test exhaustively for all available compilers, we need to build and
+# to test the C API with a different compiler each time.
+#
+# That's exactly what `test-capi` does: it runs `test-capi-*` rules
+# that, prior to testing, builds the C API with `build-capi-*` sibling
```

Comment:
I think the explanation would be clearer if it mentioned the steps it does in the order it did them, rather than #2 runs test-capi-* but first #1 it runs build-capi-*.

Reply:
Agreed. Done!